vllm/v1/core/sched/scheduler.py

```diff
@@ -645,10 +647,16 @@ def update_from_output(
                 new_running.append(request)
 
         self.running = new_running
-        return EngineCoreOutputs(
+        engine_core_outputs = EngineCoreOutputs(
             outputs=outputs,
             scheduler_stats=self.make_stats(),
         )
+        if self.include_finished_set:
```

Comment:
I think that `include_finished_set` warrants a more detailed comment about why / what this is doing